### PR TITLE
kubeadm: skip unmount on missing /var/lib/kubelet

### PR DESCRIPTION
**Pull request by neolit123**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

If the user deletes the /var/lib/kubelet manually, "reset" will throw an error that the dir is missing. Instead of handling this error, print it as a warning and skip unmount of directories inside it.

This allows "reset" to continue to be idempotent and can be called even if "init/join" are not called yet and some of the k8s directories on a node do not exist.

Continue to error on individual unmount errors.

#### Which issue(s) this PR fixes:

Follow up on https://github.com/kubernetes/kubernetes/pull/122530
xref https://github.com/kubernetes/kubeadm/issues/2923
xref https://github.com/kubernetes/kubernetes/issues/122632

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go [hunk 1]
```diff
@@ -135,7 +138,7 @@ func runCleanupNode(c workflow.RunData) error {
 func absoluteKubeletRunDirectory() (string, error) {
```
Comment (pacoxu) [inline cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go:138]: Is this function still needed?

`absoluteKubeletRunDirectory()` == `filepath.EvalSymlinks(kubeadmconstants.KubeletRunDirectory)`.

**Review by pacoxu — COMMENTED** [review]
The fix LGTM.

Not sure if the function is still needed.

File: cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go]: updated.

**Comment by pacoxu** [comment]
/lgtm

This needs to be backported with https://github.com/kubernetes/kubernetes/pull/122530 if no regression.

**Comment by neolit123** [comment]
> This needs to be backported with https://github.com/kubernetes/kubernetes/pull/122530 if no regression.

it can be backported, yes.
the commit from this PR and commit https://github.com/kubernetes/kubernetes/pull/122530/commits/d6bfd7daeb886ba9b4f273138319289646c57da3
 can be combined in a single backportable commit. i can help with reviewing / approving that.

but i would leave this task to others if they wish to: cc @mowangdk @huww98

**Pull request merged**